src/data/polynomial/ring_division.lean

```diff
@@ -681,6 +687,14 @@ begin
   exact hrew.symm
 end
 
+lemma leading_coeff_div_by_monic_X_sub_C (p : polynomial R) (hp : 0 < degree p) (a : R) :
+  leading_coeff (p /ₘ (X - C a)) = leading_coeff p :=
+begin
+  nontriviality,
+  refine leading_coeff_div_by_monic_of_monic (monic_X_sub_C a) _,
+  rwa [degree_X_sub_C, nat.with_bot.one_le_iff_zero_lt]
+end
```

Comment:
```suggestion
lemma leading_coeff_div_by_monic_X_sub_C (p : polynomial R) (hp : degree p ≠ 0) (a : R) :
  leading_coeff (p /ₘ (X - C a)) = leading_coeff p :=
begin
  nontriviality,
  cases hp.lt_or_lt with hd hd,
  { rw [degree_eq_bot.mp $ (nat.with_bot.lt_zero_iff _).mp hd, zero_div_by_monic] },
  refine leading_coeff_div_by_monic_of_monic (monic_X_sub_C a) _,
  rwa [degree_X_sub_C, nat.with_bot.one_le_iff_zero_lt]
end
```

slightly loosen requirements

Reply:
Most other lemmas of this sort ask for `0 < degree p`. The case where `degree p < 0` just means that `p = 0`, which is covered by the fact that `0 /ₘ q = 0` anyway.

Do you think we should also change `leading_coeff_div_by_monic_of_monic` to loosen it?

Reply:
how could we change that one without adding more arguments? I just think this will help reduce `eq_or_ne p 0` stuff across the library, if it comes up.